api/v1alpha1/openstackrouter_types.go

```diff
@@ -0,0 +1,177 @@
+	// specify a project identifier other than its own.
+	TenantID string `json:"tenantID,omitempty"`
+
+	// ProjectID is the project owner of the router.
+	ProjectID string `json:"projectID,omitempty"`
+
+	// All the subnets that will be added as router interfaces
+	Subnets []string `json:"subnets,omitempty"`
+
+	// Unmanaged, when true, means that no action will be performed in
+	// OpenStack against this resource. This is false by default, except
+	// for pre-existing resources that are adopted by passing ID on
+	// creation.
+	Unmanaged *bool `json:"unmanaged,omitempty"`
+}
+
+// OpenStackRouterStatus defines the observed state of OpenStackRouter
+type OpenStackRouterStatus struct {
+	// Status indicates whether or not a router is currently operational.
+	Status string `json:"status,omitempty"`
+
+	// GatewayInfo provides information on external gateway for the router.
+	GatewayInfo GatewayInfo `json:"externalGatewayInfo,omitempty"`
+
+	// InterfacesInfo provides information on the interfaces connected to this router
+	InterfacesInfo []InterfaceInfo `json:"interfacesInfo,omitempty"`
+
+	// AdminStateUp is the administrative state of the router.
+	AdminStateUp bool `json:"adminStateUp,omitempty"`
+
+	// Distributed is whether router is disitrubted or not.
+	Distributed bool `json:"distributed,omitempty"`
+
+	// Name is the human readable name for the router. It does not have to be
+	// unique.
+	Name string `json:"name,omitempty"`
+
+	// Description for the router.
+	Description string `json:"description,omitempty"`
+
+	// ID is the unique identifier for the router.
+	ID string `json:"ID,omitempty"`
+
+	// TenantID is the project owner of the router. Only admin users can
+	// specify a project identifier other than its own.
+	TenantID string `json:"tenantID,omitempty"`
+
+	// ProjectID is the project owner of the router.
+	ProjectID string `json:"projectID,omitempty"`
+
+	// Routes are a collection of static routes that the router will host.
+	Routes []Route `json:"routes,omitempty"`
+
+	// Availability zone hints groups network nodes that run services like DHCP, L3, FW, and others.
+	// Used to make network resources highly available.
+	AvailabilityZoneHints []string `json:"availabilityZoneHints,omitempty"`
+
+	// Tags optionally set via extensions/attributestags
+	Tags []string `json:"tags,omitempty"`
+}
+
+// GatewayInfo represents the information of an external gateway for any
+// particular network router.
+type GatewayInfo struct {
+	NetworkID        string            `json:"networkID,omitempty"`
+	EnableSNAT       *bool             `json:"enableSNAT,omitempty"`
+	ExternalFixedIPs []ExternalFixedIP `json:"externalFixedIps,omitempty"`
+}
+
+// ExternalFixedIP is the IP address and subnet ID of the external gateway of a
+// router.
+type ExternalFixedIP struct {
+	IPAddress string `json:"ipAddress,omitempty"`
+	SubnetID  string `json:"subnetID"`
+}
+
+// Route is a possible route in a router.
+type Route struct {
```

Comment:
Please namespace this type under the name of the resource:

```suggestion
type OpenStackRouterRoute struct {
```